suite-native/device-manager/src/components/DeviceManagerContent.tsx

```diff
@@ -1,90 +1,91 @@
+        dispatch(selectDeviceThunk(device));
+        setIsChangeDeviceRequested(false);
         setIsDeviceManagerVisible(false);
-        navigation.navigate(RootStackRoutes.ConnectDeviceStack, {
-            screen: ConnectDeviceStackRoutes.ConnectAndUnlockDevice,
-        });
+
         analytics.report({
             type: EventType.DeviceManagerClick,
-            payload: { action: 'connectDeviceButton' },
+            payload: {
+                action:
+                    device.id === PORTFOLIO_TRACKER_DEVICE_ID
+                        ? 'portfolioTracker'
+                        : 'connectDeviceButton',
+            },
         });
     };
 
-    const listedDevice = useMemo(
-        () => devices.filter(device => device.id !== selectedDeviceId),
-        [devices, selectedDeviceId],
-    );
-
     return (
-        <DeviceManagerModal>
-            {!isPortfolioTrackerDevice && <DeviceControlButtons />}
-            {!isPortfolioTrackerDevice && isPassphraseFeatureEnabled && <AddHiddenWalletButton />}
-            {A.isNotEmpty(listedDevice) && (
-                <VStack>
-                    <Text variant="callout">
-                        <Translation id="deviceManager.deviceList.sectionTitle" />
-                    </Text>
-                    {listedDevice.map(device => (
-                        <DeviceItem key={device.path} id={device.id} />
-                    ))}
-                </VStack>
-            )}
-            {isNoPhysicalDeviceConnected && (
-                <VStack>
-                    <Text variant="callout">
-                        <Translation id="deviceManager.connectDevice.sectionTitle" />
-                    </Text>
-                    <Button colorScheme="tertiaryElevation1" onPress={handleConnectDevice}>
-                        <Translation id="deviceManager.connectDevice.connectButton" />
-                    </Button>
-                </VStack>
-            )}
+        <DeviceManagerModal
+            customSwitchRightView={
+                !isPortfolioTrackerDevice && (
+                    <DevicesToggleButton
+                        deviceButtonState={isChangeDeviceRequested ? 'open' : 'closed'}
+                        onDeviceButtonTap={toggleIsChangeDeviceRequested}
+                    />
+                )
+            }
+            onClose={() => setIsChangeDeviceRequested(false)}
+        >
+            <VStack spacing="medium">
+                <DeviceList
+                    isVisible={isChangeDeviceRequested || isPortfolioTrackerDevice}
+                    onSelectDevice={handleSelectDevice}
+                />
+                {!isPortfolioTrackerDevice && (
+                    <>
+                        <WalletList onSelectDevice={handleSelectDevice} />
+                        <HStack style={applyStyle(deviceButtonsStyle)}>
+                            <DeviceInfoButton />
+                            {isPassphraseFeatureEnabled && !isDiscoveryActive && (
```

Comment:
We can't add passphrase if device is not connected. Ideally I would imagine to keep the button here but if user click it without connected device, show some reasonable warning guiding user to connect Trezor... Hiding this button would be of course easier solution, not sure how much worse from UX perspective.

Reply:
i'll hide this when device is disconnected and add issue to resolve this later after we discuss the best solution with product